airflow/models/serialized_dag.py

```diff
@@ -76,6 +82,7 @@ def __init__(self, dag: DAG):
         self.fileloc_hash = DagCode.dag_fileloc_hash(self.fileloc)
         self.data = SerializedDAG.to_dict(dag)
         self.last_updated = timezone.utcnow()
+        self.dag_hash = hashlib.md5(json.dumps(self.data, sort_keys=True).encode("utf-8")).hexdigest()
```

Comment:
I believe it will be better to use  hash of the file itself rather than Json representation. Not only it will be faster (you d note have to parse the file and create a SerializedDAG, but also it will be more accurate (there are some changes like comments that might cause the Serialized DAG to be the same (comments, jinja templates etc.). Of course it will be still ok (we care about Serialized representation in this case) but I think it would be more accurate to refresh Serialized DAG and hash every time when the file changes. Also it wil be much easier to reason about that hash and verify if it is correct or whether the file changed since - we just need to check the hash of the file on disk.

Reply:
The main reason for using Hash of a Serialized DAG is related to the question I asked Airbnb folks:

![image](https://user-images.githubusercontent.com/8811558/89730618-39257000-da38-11ea-828f-076cff27102e.png)

>Question for cong zhu: Do you store Hash of the DAG File? What happens when the modules imported by the DAG File changes bu not the DAG files itself, in that case DAG file hash remains the same but DAG might not be the same
>Cong Zhu: Yes, we store the hash of the serialized DAG

Reply:
i.e. I have seen cases where people import Schedule Interval, operators and other properties from a different file. If they change the imported file, the Hash of the DAG file remains the same but Serialized DAG changes

Reply:
Ah Right! Makes sense indeed. Great to get some input from AirBnB folks on that one :)